indico/htdocs/sass/base/_layout.scss

```diff
@@ -137,22 +136,64 @@ body > * {
     }
 }
 
-.flex {
-    display: flex;
+/* Flexbox layout */
+.flexrow {
+    flex-flow: row nowrap;
+}
+
+.flexcol {
+    flex-flow: column nowrap;
+}
+
+.flexrow,
+.flexcol {
+    display: flex !important;
+
+    &.f-j-start {
+        justify-content: flex-start;
+    }
+
+    &.f-j-end {
+        justify-content: flex-end;
+    }
+
+    &.f-j-center {
+        justify-content: center;
+    }
 
-    &.space-between {
+    &.f-j-space-between {
         justify-content: space-between;
     }
-}
 
-.stretch {
-    flex: 1;
-}
+    &.f-j-space-around {
+        justify-content: space-around;
+    }
+
+    &.f-a-start {
+        align-items: flex-start;
+    }
+
+    &.f-a-end {
+        align-items: flex-end;
+    }
+
+    &.f-a-center {
+        align-items: center;
+    }
 
-.flex-right {
-    justify-content: flex-end;
+    &.f-a-stretch {
+        align-items: stretch
+    }
+
+    &.f-a-baseline {
+        align-items: baseline;
+    }
+
+    &.f-wrap {
+        flex-wrap: wrap;
+    }
 }
 
-.flex-center {
-    justify-content: center;
+.stretch {
```

Comment:
How about giving the prefix `f-self-` to all children-related properties?
So it would be `f-self-grow`.